semantics/executable-spec/src/Control/Iterate/SetAlgebra.hs

```diff
@@ -137,22 +149,43 @@ data BiMap v a b where MkBiMap:: (v ~ b) => !(Map.Map a b) -> !(Map.Map b (Set.S
                                 --  ^   the 1st and 3rd parameter must be the same:             ^   ^
 
 -- ============== begin necessary Cardano.Binary instances ===============
-instance (Ord a, Ord b,ToCBOR a, ToCBOR b) => ToCBOR (BiMap b a b) where
-  toCBOR (MkBiMap l r) = encodeListLen 2 <> toCBOR l <> toCBOR r
-
-instance (Ord a, Ord b,FromCBOR a, FromCBOR b) => FromCBOR (BiMap b a b) where
-  fromCBOR = do
-    decodeListLenOf 2
-    !x <- fromCBOR
-    !y <- fromCBOR
-    return (MkBiMap x y)
+instance (Ord a, Ord b, ToCBOR a, ToCBOR b) => ToCBOR (BiMap b a b) where
+  -- The `toCBOR` instance encodes only the forward map. We wrap this in a
+  -- length-one list because a _previous_ encoding wrote out both maps, and we
+  -- can easily use the list length token to distinguish between them.
+  toCBOR (MkBiMap l _) = encodeListLen 1 <> toCBOR l
```

Comment:
We had to do a similar thing in consensus once. I removed that extra `encodeListLen 1` the next time the serialisation had to change. We can keep this in mind the next time the ledger format has to change.